app/uk/gov/hmrc/thirdpartydeveloperfrontend/controllers/Details.scala

```diff
@@ -205,4 +208,51 @@ class Details @Inject() (
 
   private def errorView(id: ApplicationId, form: Form[EditApplicationForm], applicationViewModel: ApplicationViewModel)(implicit request: ApplicationRequest[_]): Future[Result] =
     Future.successful(BadRequest(changeDetailsView(form, applicationViewModel)))
+
+  def canChangeProductionDetailsAndIsApprovedAction(applicationId: ApplicationId)(fun: ApplicationRequest[AnyContent] => Future[Result]): Action[AnyContent] =
+    checkActionForApprovedApps(SupportsDetails, ProductionAndAdmin)(applicationId)(fun)
+
+  def requestChangeOfAppName(applicationId: ApplicationId): Action[AnyContent] = canChangeProductionDetailsAndIsApprovedAction(applicationId) { implicit request =>
+      Future.successful(Ok(requestChangeOfApplicationNameView(ChangeOfApplicationNameForm.withData(request.application.name), ApplicationNameModel(request.application))))
+  }
+
+  def requestChangeOfAppNameAction(applicationId: ApplicationId): Action[AnyContent] = canChangeProductionDetailsAndIsApprovedAction(applicationId) { implicit request =>
+      val application = request.application
+
+      def handleValidForm(form: ChangeOfApplicationNameForm): Future[Result] = {
+        val requestForm = ChangeOfApplicationNameForm.form.bindFromRequest
+        val newApplicationName = form.applicationName
+
+        if (newApplicationName.equalsIgnoreCase(application.name)) {
```

Comment:
We could also trim whitespace from the names before comparing them?

Reply:
Good point - fixed